### PR TITLE
[REVIEW] Add negative value support for `.str.get`

### PR DESCRIPTION
**Pull request by galipremsagar**
Resolves: #3115

### CODE REVIEW

File: python/cudf/cudf/tests/test_string.py [hunk 1]
```diff
@@ -978,12 +978,19 @@ def test_string_no_children_properties():
...
     pds = pd.Series(string)
     gds = Series(string)
 
-    assert_eq(pds.str.get(index).fillna(""), gds.str.get(index).fillna(""))
+    # TODO: Remove .to_pandas.str.replace("\x00","")
+    # after this issue is fixed:https://github.com/rapidsai/cudf/issues/4515
+    assert_eq(
+        pds.str.get(index).fillna(""),
+        gds.str.get(index).fillna("").to_pandas().str.replace("\x00", ""),
+    )
```
Comment (kkraus14) [inline python/cudf/cudf/tests/test_string.py]: When we hit this case of an empty string being the expected return do we end up with the offsets column indicating nonzero sized strings as well?

Comment (galipremsagar) [reply python/cudf/cudf/tests/test_string.py]: Both offset and null count indicate 0, here is a sample I checked:
```python
>>> import cudf
>>> s = cudf.Series(["abc", ""])
>>> s.str.get(-1)
0    c
1    
dtype: object
>>> s.str.get(-1)._data.columns
(<cudf.core.column.string.StringColumn object at 0x7f98d89f5140>,)
>>> s.str.get(-1)._data.columns[0]
<cudf.core.column.string.StringColumn object at 0x7f98d89f5af0>
>>> s.str.get(-1)._data.columns[0].null_count
0
>>> s.str.get(-1)._data.columns[0]._offset
0
>>> s.str.get(-1)._data.columns[0].to_array()
array(['c', '\x00'], dtype=object)
>>> s._data.columns[0].to_array()
array(['abc', ''], dtype=object)
```

Comment (kkraus14) [reply python/cudf/cudf/tests/test_string.py]: @galipremsagar I was referring to `s.str.get(-1)._data.columns[0].children[0]` and `s.str.get(-1)._data.columns[0].children[1]`

Comment (davidwendt) [reply python/cudf/cudf/tests/test_string.py]: I can fix #4515 if that helps. It looks to be a simple fix.

Comment (galipremsagar) [reply python/cudf/cudf/tests/test_string.py]: Yep, that should help. Then holding this PR until that fix is merged should be fine? @kkraus14

Comment (galipremsagar) [reply python/cudf/cudf/tests/test_string.py]: @davidwendt Thanks for the fix(#4538 ). 🎉 The issue is now gone.
@kkraus14 This PR is now unblocked.

**Pull request merged**